include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -505,8 +510,13 @@ struct __get_sycl_range
         auto __n = __last - __first;
         assert(__n > 0);
 
-        auto res_src =
-            __process_input_iter<_LocalAccMode>(__first.base(), oneapi::dpl::end(__first.base().get_buffer()));
+        // Types for which oneapi::dpl::__ranges::is_sycl_iterator_v = true should have both:
+        //  "get_buffer()" to return the buffer they are base upon and
```

Comment:
There is no explanation in the comments why we need new code.

Reply:
comment added.